app/models/manageiq/providers/redfish/physical_infra_manager/operations/power.rb

```diff
@@ -0,0 +1,65 @@
+module ManageIQ::Providers::Redfish
+  module PhysicalInfraManager::Operations::Power
+    # Keep this in sync with app/models/physical_server/operations/power.rb in
+    # core and ResetType enum in Redfish Resource type. Name of the method
+    # comes from the core and the action name used in the reset call from the
+    # ResetType enum.
+    #
+    # NOTE: Not all reset operations are implemented on all servers, so any of
+    # the methods listed here can fail. We need to find a way to let those
+    # failures bubble up to the user interface somehow or risk having a
+    # completely useless tool.
+
+    def power_on(server, _options)
+      reset_server(server, "On")
+    end
+
+    def power_off(server, _options)
+      reset_server(server, "GracefulShutdown")
+    end
+
+    def power_off_now(server, _options)
+      reset_server(server, "ForceOff")
+    end
+
+    def restart(server, _options)
+      reset_server(server, "GracefulRestart")
+    end
+
+    def restart_now(server, _options)
+      reset_server(server, "ForceRestart")
+    end
+
+    def restart_to_sys_setup(_args, _options)
+      $redfish_log.error("Restarting to system setup is not supported.")
```

Comment:
Hm we should have a supports feature for this so the UI/API can check if the method is supported before calling it, something to do in a followup as I'm not sure any of that exists for physical server power ops.